### DISCUSSION TITLE
Disable streaming rendering for google bots and clients that do not support http2 streaming

### DISCUSSION THREAD

**Discussion by Fredkiss3** [vercel/next.js#43505] [Ideas]
### Describe the feature you'd like to request

Referencing this discussion on the documentation https://beta.nextjs.org/docs/api-reference/notfound#thread-id=qPXdO, when we use `notFound` on a server component and inspect the result on google bot or a simple crawler the response returned is a `200 OK`, while it should maybe return a `404`.   

### Describe the solution you'd like

Before serving the request, Nextjs could inspect the user-agent of the request and if the agent cannot support streaming :

- `notFound` should return a `404 Not Found` response code 
-  And `redirect` should return a `302 Found` response code 

### Describe alternatives you've considered

No alternatives yet.

**Comment by balazsorban44** [comment]
Hi, this is just not implemented yet. :+1:

**Reply by Fredkiss3** [reply]
It's is the roadmap i suppose ?

**Comment by Sharpiro** [comment]
Are there any issues tracking this yet?

**Comment by Fredkiss3** [comment]
I think the canonical solution here is to use `notFound` & `redirect` in `generateMetadata()` there it sets the correct HTTP codes.

> The [redirect()](https://nextjs.org/docs/app/api-reference/functions/redirect) and [notFound()](https://nextjs.org/docs/app/api-reference/functions/not-found) Next.js methods can also be used inside generateMetadata.

from the docs : https://nextjs.org/docs/app/api-reference/functions/generate-metadata#returns

**Reply by steebchen** [reply]
This is a great temporary workaround but I don't believe it is the proper solution. My biggest concern is that server errors aren't caught this way, and could even go completely unnoticed. Say google scrapes your site, but your data fetching fails; it would always return 200 OK with an error being rendered there which Google would try to index. In this case it's important to return a 500 so Google doesn't index and re-tries instead. I think it would be better to re-open the discussion.

**Reply by steebchen** [reply]
Plus, it looks like this doesn't work in all cases right now: https://github.com/vercel/next.js/issues/60351

**Comment by steebchen** [comment]
Another issue is that any server error will not return a 500, but always a 200. I guess there's not really a fix as it's just how streaming works, but I wanted to add this here as a good-to-know. In my opinion, this feature request should be re-opened.

**Reply by guillaume-fr** [reply]
Agreed. Streaming can be nice and maybe you want that in some case but I'd like to have an option to enable: 
- plain old SSR, typically for bots with no JS support
- separate requests for dynamic content so each may generate its HTTP response. Useful when your architecture requires interoperability with other solutions (security solutions, reverse proxy, observability/logs)

**Comment by zubkov7** [comment]
Hi! Is there any updates or plans to integrate in future versions?
